### PR TITLE
 pkgs/development: switch to fetchFromGitHub part 4

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
use `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/lwt/default.nix [hunk 1]
```diff
@@ -12,9 +12,11 @@ buildDunePackage rec {
 
   useDune2 = true;
 
-  src = fetchzip {
-    url = "https://github.com/ocsigen/${pname}/archive/${version}.tar.gz";
-    sha256 = "0cq2qy23sa1a5zk6nja3c652mp29i84yfrkcwks6i8sdqwli36jy";
+  src = fetchFromGitHub {
+    owner = "ocsigen";
+    repo = "lwt";
+    rev = "5.4.1";
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/lwt/default.nix:18]:
```suggestion
    rev = version;
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 143900` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>9 packages failed to build and are new build failures:</summary>
  <ul>
<li>comby: log was empty</li>
<li>coqPackages.serapi: log was empty</li>
<li>dot-merlin-reader: log was empty</li>
<li>dune-release: log was empty</li>
<li>eff: log was empty</li>
<li>fstar: log was empty</li>
<li>haxe_4_0: log was empty</li>
<li>sks: log was empty</li>
<li>stog: log was empty</li>
  </ul>
</details>
<details>
<summary>
34 packages built:
</summary>
<ul>
<li>acgtk</li>
<li>framac</li>
<li>google-drive-ocamlfuse</li>
<li>haxe (haxe_4_2)</li>
<li>hxcpp (haxePackages.hxcpp)</li>
<li>haxePackages.hxcs</li>
<li>haxePackages.hxjava</li>
<li>haxePackages.hxnodejs_4</li>
<li>haxe_4_1</li>
<li>jackline</li>
<li>libbap</li>
<li>ligo</li>
<li>liquidsoap</li>
<li>ocaml-crunch</li>
<li>ocamlformat_0_11_0</li>
<li>ocamlformat_0_12</li>
<li>ocamlformat_0_13_0</li>
<li>ocamlformat_0_14_0</li>
<li>ocamlformat_0_14_1</li>
<li>ocamlformat_0_14_2</li>
<li>ocamlformat_0_14_3</li>
<li>ocamlformat_0_15_0</li>
<li>ocamlformat_0_15_1</li>
<li>ocamlformat_0_16_0</li>
<li>ocamlformat_0_17_0</li>
<li>ocamlformat_0_18_0</li>
<li>opa</li>
<li>python38Packages.bap</li>
<li>python39Packages.bap</li>
<li>satysfi</li>
<li>sawjap</li>
<li>soupault</li>
<li>wayland-proxy-virtwl</li>
<li>why3</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
comby:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/build-support/ocaml/dune.nix:40:3:

```
   |
40 |   meta = (args.meta or {}) // { platforms = args.meta.platforms or ocaml.meta.platforms; };
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
eff:
<details>


warning: build-tools-in-build-inputs
which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/interpreters/eff/default.nix:18:3:

```
   |
18 |   buildInputs = [ which ] ++ (with ocamlPackages; [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
framac:
<details>

error: build log of '/nix/store/bai3j3ss1jzpybv5k0fv79ldz68nchk4-frama-c-23.1' is not available
When evaluating attribute ‘framac’:
warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/tools/analysis/frama-c/default.nix:83:5:

```
   |
83 |     license     = lib.licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
doxygen is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/analysis/frama-c/default.nix:46:3:

```
   |
46 |   buildInputs = with ocamlPackages; [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: unused-argument
Unused argument: `makeWrapper`.
Near pkgs/development/tools/analysis/frama-c/default.nix:1:26:

```
  |
1 | { lib, stdenv, fetchurl, makeWrapper, writeText
  |                          ^
```
</details>
haxePackages.hxcs:
<details>

error: build log of '/nix/store/wan16r57i0x1zj6whbd108xq79nhvswd-hxcs-3.4.0' is not available
When evaluating attribute ‘haxePackages.hxcs’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:86:7:

```
   |
86 |       version = "3.4.0";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
haxePackages.hxjava:
<details>

error: build log of '/nix/store/rippp9pbik4kbpswhj85cjw48srk2wdd-hxjava-3.2.0' is not available
When evaluating attribute ‘haxePackages.hxjava’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:78:7:

```
   |
78 |       version = "3.2.0";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
haxePackages.hxnodejs_4:
<details>

error: build log of '/nix/store/05i7q5zmimiknaqgcpp8mllxkkcj1z5k-hxnodejs-4.0.9' is not available
When evaluating attribute ‘haxePackages.hxnodejs_4’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:94:7:

```
   |
94 |       version = "4.0.9";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
hxcpp:
<details>

error: build log of '/nix/store/ff8h5469yrpby0sgyiv8lvbksyklbs73-hxcpp-4.1.15' is not available
When evaluating attribute ‘hxcpp’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:64:7:

```
   |
64 |       version = "4.1.15";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
jackline:
<details>

error: build log of '/nix/store/yr6zv6mcdjz0rwhhrqanllc167fjg5fm-ocaml4.08.1-jackline-unstable-2021-08-10' is not available
When evaluating attribute ‘jackline’:
warning: attribute-typo
A likely typo in the `nativeBuildInpts` argument was found, did you mean `nativeBuildInputs`?

Near pkgs/applications/networking/instant-messengers/jackline/default.nix:20:3:

```
   |
20 |   nativeBuildInpts = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
</details>
liquidsoap:
<details>

error: build log of '/nix/store/5dpyi74q4nzg3qg7wrkmvqrf8lyjzwxz-liquidsoap-full-1.4.2' is not available
When evaluating attribute ‘liquidsoap’:
warning: unclear-gpl
`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/audio/liquidsoap/full.nix:71:5:

```
   |
71 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/audio/liquidsoap/full.nix:53:3:

```
   |
53 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
opa:
<details>

error: build log of '/nix/store/695x5ba84z261vck0pxxpds020wmlqgg-opa-4310' is not available
When evaluating attribute ‘opa’:
warning: unclear-gpl
`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/compilers/opa/default.nix:74:5:

```
   |
74 |     license = lib.licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/compilers/opa/default.nix:47:3:

```
   |
47 |   buildInputs = [ which perl jdk openssl coreutils zlib ncurses
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/compilers/opa/default.nix:47:3:

```
   |
47 |   buildInputs = [ which perl jdk openssl coreutils zlib ncurses
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
error: no-flags-spaces
`configureFlags` cannot contain spaces, please use `configureFlagsArray` in Bash.

Near pkgs/development/compilers/opa/default.nix:45:3:

```
   |
45 |   configureFlags = [ "-ocamlfind ${ocamlPackages.findlib}/bin/ocamlfind" ];
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-spaces.md
</details>
python38Packages.bap:
<details>

error: build log of '/nix/store/ik0sz1kbsksn9rmfsysrwvgmz3h53xfz-python3.8-bap-1.3.1' is not available
When evaluating attribute ‘python38Packages.bap’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/bap/default.nix:18:

```
   |
18 |     description = "Platform for binary analysis. It is written in OCaml, but can be used from other languages.";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.bap:
<details>

error: build log of '/nix/store/q31f9qx6xyxndjam86q0av086yk23i6n-python3.9-bap-1.3.1' is not available
When evaluating attribute ‘python39Packages.bap’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/bap/default.nix:18:

```
   |
18 |     description = "Platform for binary analysis. It is written in OCaml, but can be used from other languages.";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
stog:
<details>


warning: unclear-gpl
`lgpl3` is a deprecated license, please check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/stog/default.nix:26:5:

```
   |
26 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
why3:
<details>

error: build log of '/nix/store/3r9vysx55k8svw6ashwz60kn6dm52sxp-why3-1.4.0' is not available
When evaluating attribute ‘why3’:
warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/applications/science/logic/why3/default.nix:42:5:

```
   |
42 |     license     = licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: unused-argument
Unused argument: `fetchpatch`.
Near pkgs/applications/science/logic/why3/default.nix:1:26:

```
  |
1 | { callPackage, fetchurl, fetchpatch, lib, stdenv
  |                          ^
```
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 143900` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>6 packages failed to build and are new build failures:</summary>
  <ul>
<li>comby: log was empty</li>
<li>coqPackages.serapi: log was empty</li>
<li>dune-release: log was empty</li>
<li>eff: log was empty</li>
<li>fstar: log was empty</li>
<li>stog: log was empty</li>
  </ul>
</details>
<details>
<summary>
37 packages built:
</summary>
<ul>
<li>acgtk</li>
<li>dot-merlin-reader</li>
<li>framac</li>
<li>google-drive-ocamlfuse</li>
<li>haxe (haxe_4_2)</li>
<li>hxcpp (haxePackages.hxcpp)</li>
<li>haxePackages.hxcs</li>
<li>haxePackages.hxjava</li>
<li>haxePackages.hxnodejs_4</li>
<li>haxe_4_0</li>
<li>haxe_4_1</li>
<li>jackline</li>
<li>libbap</li>
<li>ligo</li>
<li>liquidsoap</li>
<li>ocaml-crunch</li>
<li>ocamlformat_0_11_0</li>
<li>ocamlformat_0_12</li>
<li>ocamlformat_0_13_0</li>
<li>ocamlformat_0_14_0</li>
<li>ocamlformat_0_14_1</li>
<li>ocamlformat_0_14_2</li>
<li>ocamlformat_0_14_3</li>
<li>ocamlformat_0_15_0</li>
<li>ocamlformat_0_15_1</li>
<li>ocamlformat_0_16_0</li>
<li>ocamlformat_0_17_0</li>
<li>ocamlformat_0_18_0</li>
<li>opa</li>
<li>python38Packages.bap</li>
<li>python39Packages.bap</li>
<li>satysfi</li>
<li>sawjap</li>
<li>sks</li>
<li>soupault</li>
<li>wayland-proxy-virtwl</li>
<li>why3</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
comby:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/build-support/ocaml/dune.nix:40:3:

```
   |
40 |   meta = (args.meta or {}) // { platforms = args.meta.platforms or ocaml.meta.platforms; };
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
eff:
<details>


warning: build-tools-in-build-inputs
which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/interpreters/eff/default.nix:18:3:

```
   |
18 |   buildInputs = [ which ] ++ (with ocamlPackages; [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
framac:
<details>

error: build log of '/nix/store/bai3j3ss1jzpybv5k0fv79ldz68nchk4-frama-c-23.1' is not available
When evaluating attribute ‘framac’:
warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/tools/analysis/frama-c/default.nix:83:5:

```
   |
83 |     license     = lib.licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
doxygen is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/analysis/frama-c/default.nix:46:3:

```
   |
46 |   buildInputs = with ocamlPackages; [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: unused-argument
Unused argument: `makeWrapper`.
Near pkgs/development/tools/analysis/frama-c/default.nix:1:26:

```
  |
1 | { lib, stdenv, fetchurl, makeWrapper, writeText
  |                          ^
```
</details>
haxePackages.hxcs:
<details>

error: build log of '/nix/store/wan16r57i0x1zj6whbd108xq79nhvswd-hxcs-3.4.0' is not available
When evaluating attribute ‘haxePackages.hxcs’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:86:7:

```
   |
86 |       version = "3.4.0";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
haxePackages.hxjava:
<details>

error: build log of '/nix/store/rippp9pbik4kbpswhj85cjw48srk2wdd-hxjava-3.2.0' is not available
When evaluating attribute ‘haxePackages.hxjava’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:78:7:

```
   |
78 |       version = "3.2.0";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
haxePackages.hxnodejs_4:
<details>

error: build log of '/nix/store/05i7q5zmimiknaqgcpp8mllxkkcj1z5k-hxnodejs-4.0.9' is not available
When evaluating attribute ‘haxePackages.hxnodejs_4’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:94:7:

```
   |
94 |       version = "4.0.9";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
hxcpp:
<details>

error: build log of '/nix/store/ff8h5469yrpby0sgyiv8lvbksyklbs73-hxcpp-4.1.15' is not available
When evaluating attribute ‘hxcpp’:
warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/top-level/haxe-packages.nix:31:9:

```
   |
31 |         name = "${libname}-${version}";
   |         ^
```

Near pkgs/top-level/haxe-packages.nix:64:7:

```
   |
64 |       version = "4.1.15";
   |       ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/top-level/haxe-packages.nix:54:9:

```
   |
54 |         meta = {
   |         ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
jackline:
<details>

error: build log of '/nix/store/yr6zv6mcdjz0rwhhrqanllc167fjg5fm-ocaml4.08.1-jackline-unstable-2021-08-10' is not available
When evaluating attribute ‘jackline’:
warning: attribute-typo
A likely typo in the `nativeBuildInpts` argument was found, did you mean `nativeBuildInputs`?

Near pkgs/applications/networking/instant-messengers/jackline/default.nix:20:3:

```
   |
20 |   nativeBuildInpts = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
</details>
liquidsoap:
<details>

error: build log of '/nix/store/5dpyi74q4nzg3qg7wrkmvqrf8lyjzwxz-liquidsoap-full-1.4.2' is not available
When evaluating attribute ‘liquidsoap’:
warning: unclear-gpl
`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/audio/liquidsoap/full.nix:71:5:

```
   |
71 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/audio/liquidsoap/full.nix:53:3:

```
   |
53 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
opa:
<details>

error: build log of '/nix/store/695x5ba84z261vck0pxxpds020wmlqgg-opa-4310' is not available
When evaluating attribute ‘opa’:
warning: unclear-gpl
`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/compilers/opa/default.nix:74:5:

```
   |
74 |     license = lib.licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/compilers/opa/default.nix:47:3:

```
   |
47 |   buildInputs = [ which perl jdk openssl coreutils zlib ncurses
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
warning: build-tools-in-build-inputs
which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/compilers/opa/default.nix:47:3:

```
   |
47 |   buildInputs = [ which perl jdk openssl coreutils zlib ncurses
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
error: no-flags-spaces
`configureFlags` cannot contain spaces, please use `configureFlagsArray` in Bash.

Near pkgs/development/compilers/opa/default.nix:45:3:

```
   |
45 |   configureFlags = [ "-ocamlfind ${ocamlPackages.findlib}/bin/ocamlfind" ];
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-spaces.md
</details>
python38Packages.bap:
<details>

error: build log of '/nix/store/ik0sz1kbsksn9rmfsysrwvgmz3h53xfz-python3.8-bap-1.3.1' is not available
When evaluating attribute ‘python38Packages.bap’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/bap/default.nix:18:

```
   |
18 |     description = "Platform for binary analysis. It is written in OCaml, but can be used from other languages.";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.bap:
<details>

error: build log of '/nix/store/q31f9qx6xyxndjam86q0av086yk23i6n-python3.9-bap-1.3.1' is not available
When evaluating attribute ‘python39Packages.bap’:
warning: python-include-tests
Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/bap/default.nix:18:

```
   |
18 |     description = "Platform for binary analysis. It is written in OCaml, but can be used from other languages.";
   | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
stog:
<details>


warning: unclear-gpl
`lgpl3` is a deprecated license, please check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/stog/default.nix:26:5:

```
   |
26 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
why3:
<details>

error: build log of '/nix/store/3r9vysx55k8svw6ashwz60kn6dm52sxp-why3-1.4.0' is not available
When evaluating attribute ‘why3’:
warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/applications/science/logic/why3/default.nix:42:5:

```
   |
42 |     license     = licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: unused-argument
Unused argument: `fetchpatch`.
Near pkgs/applications/science/logic/why3/default.nix:1:26:

```
  |
1 | { callPackage, fetchurl, fetchpatch, lib, stdenv
  |                          ^
```
</details>
</details>

**Comment by Stunkymonkey** [comment]
@SuperSandro2000 should I close this due to the fact, that OCaml is getting deprecated?

**Comment by SuperSandro2000** [comment]
If that is not right around the corner I would just merge this PR after the merge conflict is resolved. Can you give me a ping when you force push it? I try to merge it then within a few days.

**Comment by Stunkymonkey** [comment]
i just force-pushed. The conflict was on a file, that got removed.

**Comment by vbgl** [comment]
Please drop the commit touching `faillib`.

**Comment by Stunkymonkey** [comment]
> Please drop the commit touching `faillib`.

done.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 143900` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>fstar: log was empty</li>
<li>ligo: log was empty</li>
<li>ocamlformat_0_15_0: log was empty</li>
  </ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>opa: log was empty</li>
  </ul>
</details>
<details>
<summary>
39 packages built:
</summary>
<ul>
<li>acgtk</li>
<li>comby</li>
<li>coqPackages.serapi</li>
<li>dot-merlin-reader</li>
<li>dune-release</li>
<li>eff</li>
<li>framac</li>
<li>google-drive-ocamlfuse</li>
<li>haxe (haxe_4_2)</li>
<li>hxcpp (haxePackages.hxcpp)</li>
<li>haxePackages.hxcs</li>
<li>haxePackages.hxjava</li>
<li>haxePackages.hxnodejs_4</li>
<li>haxe_4_0</li>
<li>haxe_4_1</li>
<li>jackline</li>
<li>libbap</li>
<li>liquidsoap</li>
<li>ocaml-crunch</li>
<li>ocamlformat_0_11_0</li>
<li>ocamlformat_0_12</li>
<li>ocamlformat_0_13_0</li>
<li>ocamlformat_0_14_0</li>
<li>ocamlformat_0_14_1</li>
<li>ocamlformat_0_14_2</li>
<li>ocamlformat_0_14_3</li>
<li>ocamlformat_0_15_1</li>
<li>ocamlformat_0_16_0</li>
<li>ocamlformat_0_17_0</li>
<li>ocamlformat_0_18_0</li>
<li>python38Packages.bap</li>
<li>python39Packages.bap</li>
<li>satysfi</li>
<li>sawjap</li>
<li>sks</li>
<li>soupault</li>
<li>stog</li>
<li>wayland-proxy-virtwl</li>
<li>why3</li>
</ul>
</details>


---

manually build opa

**Pull request merged**